### PR TITLE
Test itself by schema_salad

### PR DESCRIPTION
**Pull request by otiai10**
- add schema_salad as a submodule
- implement `--self` framework in `run_test.sh`
- fix test case which is obviously failing

### CODE REVIEW

**Comment by stain** [comment]
I think this is a good thing, as evidenced by the [travis build](https://travis-ci.org/common-workflow-language/common-workflow-language/builds/404355242?utm_source=github_status&utm_medium=notification) already finding a problem:

```
Document `v1.0/v1.0/count-lines10-wf.cwl` failed validation:
v1.0/v1.0/count-lines10-wf.cwl:10:1:  checking field `steps`
v1.0/v1.0/count-lines10-wf.cwl:11:3:    checking object `v1.0/v1.0/count-lines10-wf.cwl#step0`
v1.0/v1.0/count-lines10-wf.cwl:12:5:      checking field `in`
v1.0/v1.0/count-lines10-wf.cwl:12:10:       checking object
                                            `v1.0/v1.0/count-lines10-wf.cwl#step0/file1`
v1.0/v1.0/count-lines10-wf.cwl:17:9:          object id
                                              `v1.0/v1.0/count-lines10-wf.cwl#step0/file1`
                                              previously defined
[INVALID] v1.0/v1.0/count-lines10-wf.cwl
```

Is it OK that it's hard-coded for v1.0, or should rather version of the schema be cross-connected to the corresponding diretory to check?

**Comment by mr-c** [comment]
Jenkins, add to testlist

File: run_test.sh [hunk 1]
```diff
@@ -65,12 +68,35 @@ do
...
+    if [[ ! -x $(command -v schema-salad-tool) ]]; then
+        if [[ ! -d ./schema_salad ]]; then
+            echo >&2 "You need: git submodule update --init"
+            exit 1
+        fi
+        # Install schema_salad to validate cwl files.
+        cd ./schema_salad && pip install . --quiet && cd ..
+    fi
+    # This is how CWL should be written.
+    DEFINITION=./schema_salad/schema_salad/tests/test_schema/CommonWorkflowLanguage.yml
```
Comment (mr-c) [inline run_test.sh]: That file is two years out of date, why not use the one in this repository?

Comment (otiai10) [reply run_test.sh]: Ah! I didn't see it. I will fix it to see the `./v1.0/CommonWorkflowLanguage.yml`!!

**Comment by otiai10** [comment]
fixed by https://github.com/common-workflow-language/common-workflow-language/pull/732/commits/2635f98330f5acc6b4ebe6ea0ddfc57530495df1

**Comment by otiai10** [comment]
I just checked [the latest version of schema_salad](https://github.com/common-workflow-language/schema_salad/releases/tag/2.7.20180724081820) fixes the test status, **at least Travis CI**, which tests the validity of cwl files.

Thank you guys for publishing.

**Comment by tom-tan** [comment]
Ping.
Are there any blockers for this request?

**Comment by tetron** [comment]
I have no issue but @mr-c needs to resolve the review comment he made.

**Comment by mr-c** [comment]
Thank you @otiai10 !

**Comment by otiai10** [comment]
@mr-c thank you too.

Then, strongly believing that keeping it green is the first attitude of OSS, I want to fix the Jenkins status as well.

Is there anything I can do for that?

**Comment by otiai10** [comment]
image

- https://ci.commonwl.org/job/cwl-spec-pr/396/
- https://ci.commonwl.org/job/cwl-spec-pr-cwlexec/44/

Because these CI are red, contributors who send pull-request can't be sure that his/her change does NOT break CWL itself.
If these CI or the tools used inside these CI (rabix v1.0 and cwlexec) are nonsense, these CI should be updated otherwise deleted, so as to encourage developers to contribute CWL ecosystem.

**Comment by mr-c** [comment]
@otiai10 Thanks for the clarification. I will remove Rabix Bunny as the latest open source code is not currently working.
CWLExec is not marked as "required", but maybe that is confusing for people

**Pull request merged**